aten/src/ATen/native/Attention.cpp

```diff
@@ -0,0 +1,58 @@
+#include <ATen/ATen.h>
+
+namespace at {
+namespace native {
+std::tuple<Tensor, Tensor> attn(
+    const Tensor& q,
+    const Tensor& k,
+    const Tensor& v) {
```

Comment:
note that in general for this, it is nice to have a couple `TORCH_CHECK` to make sure the input ndim and size are what you expect.